itests/fevm_test.go

```diff
@@ -1436,3 +1436,125 @@ func TestEthGetBlockByNumber(t *testing.T) {
 	require.NotNil(t, pendingBlock)
 	require.True(t, pendingBlock.Number >= latest)
 }
+
+func TestEthGetTransactionByBlockHashAndIndexAndNumber(t *testing.T) {
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	ethKey, ethAddr, ethFilAddr := client.EVM().NewAccount()
+	kit.SendFunds(ctx, t, client, ethFilAddr, types.FromFil(10))
+
+	var txHashes []ethtypes.EthHash
+	var receipts []*api.EthTxReceipt
+	numTx := 3
+
+	contractHex, err := os.ReadFile("./contracts/MultipleEvents.hex")
+	require.NoError(t, err)
+	contract, err := hex.DecodeString(string(contractHex))
+	require.NoError(t, err)
+
+	gasParams, err := json.Marshal(ethtypes.EthEstimateGasParams{Tx: ethtypes.EthCall{
+		From: &ethAddr,
+		Data: contract,
+	}})
+	require.NoError(t, err)
+	gaslimit, err := client.EthEstimateGas(ctx, gasParams)
+	require.NoError(t, err)
+
+	maxPriorityFeePerGas, err := client.EthMaxPriorityFeePerGas(ctx)
+	require.NoError(t, err)
+
+	for {
+		txHashes = nil
+		receipts = nil
+		nonce, err := client.MpoolGetNonce(ctx, ethFilAddr)
+		require.NoError(t, err)
+
+		for i := 0; i < numTx; i++ {
+			tx := &ethtypes.Eth1559TxArgs{
+				ChainID:              buildconstants.Eip155ChainId,
+				Value:                big.Zero(),
+				Nonce:                int(nonce) + i,
+				MaxFeePerGas:         types.NanoFil,
+				MaxPriorityFeePerGas: big.Int(maxPriorityFeePerGas),
+				GasLimit:             int(gaslimit),
+				Input:                contract,
+				V:                    big.Zero(),
+				R:                    big.Zero(),
+				S:                    big.Zero(),
+			}
+			client.EVM().SignTransaction(tx, ethKey.PrivateKey)
+			hash := client.EVM().SubmitTransaction(ctx, tx)
+			txHashes = append(txHashes, hash)
+		}
+
+		for _, hash := range txHashes {
+			receipt, err := client.EVM().WaitTransaction(ctx, hash)
+			require.NoError(t, err)
+			require.NotNil(t, receipt)
+			receipts = append(receipts, receipt)
+		}
+
+		allInSameTipset := true
+		for i := 1; i < len(receipts); i++ {
+			if receipts[i].BlockHash != receipts[0].BlockHash {
+				allInSameTipset = false
+				break
+			}
+		}
+
+		if allInSameTipset {
+			break
+		}
```

Comment:
optional, but you could `t.Logf` here to say you're looping because they didn't land in the same tipset - that may help with a pathological case of the test timing out because we couldn't get them to land (I don't know if we'd ever experience that, so it's probably unnecessary)

Reply:
I'll add `t.Logf("Retrying because transactions didn't land in the same tipset")`